airflow/providers/amazon/aws/triggers/emr.py

```diff
@@ -102,274 +102,184 @@ async def run(self):
+            status_queries=[
+                "Cluster.Status.State",
+                "Cluster.Status.StateChangeReason",
+                "Cluster.Status.ErrorDetails",
+            ],
+            return_value=None,
+            waiter_delay=waiter_delay,
+            waiter_max_attempts=waiter_max_attempts,
+            aws_conn_id=aws_conn_id,
         )
 
-    async def run(self):
-        self.hook = EmrHook(aws_conn_id=self.aws_conn_id)
-        async with self.hook.async_conn as client:
-            attempt = 0
-            waiter = self.hook.get_waiter("job_flow_terminated", deferrable=True, client=client)
-            while attempt < int(self.max_attempts):
-                attempt = attempt + 1
-                try:
-                    await waiter.wait(
-                        ClusterId=self.job_flow_id,
-                        WaiterConfig=prune_dict(
-                            {
-                                "Delay": self.poll_interval,
-                                "MaxAttempts": 1,
-                            }
-                        ),
-                    )
-                    break
-                except WaiterError as error:
-                    if "terminal failure" in str(error):
-                        raise AirflowException(f"JobFlow termination failed: {error}")
-                    self.log.info(
-                        "Status of jobflow is %s - %s",
-                        error.last_response["Cluster"]["Status"]["State"],
-                        error.last_response["Cluster"]["Status"]["StateChangeReason"],
-                    )
-                    await asyncio.sleep(int(self.poll_interval))
-        if attempt >= int(self.max_attempts):
-            raise AirflowException(f"JobFlow termination failed - max attempts reached: {self.max_attempts}")
-        else:
-            yield TriggerEvent(
-                {
-                    "status": "success",
-                    "message": "JobFlow terminated successfully",
-                }
-            )
+    def hook(self) -> AwsGenericHook:
+        return EmrHook(aws_conn_id=self.aws_conn_id)
 
 
-class EmrContainerTrigger(BaseTrigger):
+class EmrContainerTrigger(AwsBaseWaiterTrigger):
     """
     Poll for the status of EMR container until reaches terminal state.
 
     :param virtual_cluster_id: Reference Emr cluster id
     :param job_id:  job_id to check the state
     :param aws_conn_id: Reference to AWS connection id
-    :param poll_interval: polling period in seconds to check for the status
+    :param waiter_delay: polling period in seconds to check for the status
     """
 
     def __init__(
         self,
         virtual_cluster_id: str,
         job_id: str,
         aws_conn_id: str = "aws_default",
-        poll_interval: int = 30,
-        **kwargs: Any,
+        poll_interval: int | None = None,  # deprecated
+        waiter_delay: int = 30,
+        waiter_max_attempts: int = 600,
```

Comment:
is this supposed to be 60?

Reply:
it's supposed to be ∞ lol
we don't really have a clear posture on this, some triggers have sensible max attempts values, some others wait forever...
What I don't like with setting a "low" value like 60 is that if the user sets the poll interval to 1 for instance, it ends in 1 minute, which might be annoying.
Overall, I think it should be a really high value, and users should set it themselves if they care.

Reply:
I think a good compromise is to keep it the same as the default one used by boto. That way, we don't inadvertently mess things up for users who might be relying on a Task to fail after a certain amount of time, because they didn't change the default value

Reply:
ok, but in this particular case, the existing behavior was to wait forever, so....